doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,384 @@
+
+Device wires 
+------------
+
+Currently, ``default.qubit`` and ``lightning.qubit`` are compatible with program-capture.
+With program-capture enabled, both ``lightning.qubit`` and ``default.qubit`` require 
+that ``wires`` be specified at device instantiation (this is in contrast to when 
+program-capture is disabled, where automatic qubit management takes place internally
+with ``default.qubit``).
+
+.. code-block:: python
+
+    @qml.qnode(qml.device('default.qubit'))
+    def circuit():
+        qml.Hadamard(0)
+        return qml.state()
+
+>>> circuit()
+NotImplementedError: devices must specify wires for integration with plxpr capture.
+
+Valid JAX data types 
+--------------------
+
+Because of the nature of creating and executing plxpr, it is **best practice to 
+use JAX-compatible types whenever possible**, in particular for arguments to quantum 
+functions and QNodes, and positional arguments in PennyLane gate operations. JAX-compatible 
+types are ``jax.numpy`` arrays and standard Python ``int``\ s and ``float``\ s. 
+
+For example, ``list``\ s, ``range``\ s, and strings are not valid JAX types for 
+the positional argument in :class:`~.pennylane.MultiRZ`, and will result in an error:
+
+.. code-block:: python
+    dev = qml.device('default.qubit', wires=2)
+
+    @qml.qnode(dev)
+    def circuit():
+        qml.MultiRZ([0.1, 0.2], wires=[0, 1])
+        return qml.expval(qml.X(0))
+
+>>> circuit()
+TypeError: Value [0.1, 0.2] with type <class 'list'> is not a valid JAX type
+
+Providing a ``list`` as input to a quantum function or QNode is accepted in cases 
+where the ``list`` is being indexed into, thereby retrieving a valid JAX type:
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit', wires=2)
+
+    @qml.qnode(dev)
+    def circuit(x):
+        qml.RZ(x[0], wires=0)
+        qml.RX(x[1], wires=1)
+        return qml.expval(qml.X(0))
+
+>>> circuit([0.1, 0.2])
+Array(0., dtype=float32)
+
+JAX-incompatible types, like Python ``range``\ s, are acceptable as keyword arguments:
+
+.. code-block:: python
+    
+    dev = qml.device('default.qubit', wires=2)
+
+    @qml.qnode(dev)
+    def circuit(x, range_of_wires=None):
+        for w in range_of_wires:
+            qml.RZ(x[0], wires=w)
+            qml.RX(x[1], wires=w)
+
+        return qml.expval(qml.X(0))
+
+>>> circuit([0.1, 0.2], range_of_wires=range(2))
+Array(0., dtype=float32)
+
+But, again, using JAX-compatible types wherever possible is recommended.
+
+Positional arguments
+====================
+
+Positional arguments **must** be acceptable JAX types. For instance, consider this
+example with ``qml.RZ``:
+
+.. code-block:: python
+
+    import jax.numpy as jnp
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circui(angle):
+        qml.RX(phi=angle, wires=0)
+        return qml.expval(qml.Z(0))
+
+>>> angle = jnp.array(0.1)
+>>> circuit1(angle)
```

Comment:
I'm not sure I understand -- this is a valid jax type?

Reply:
Typo :)